src/espressif/buffered_uart.rs

```diff
@@ -68,6 +96,12 @@ impl BufferedUart {
         self.outward.write_all(buf).await;
     }
 
+    /// Return the number of dropped bytes (if any) since the last check,
+    /// and reset the internal count to 0.
```

Comment:
One thing to note about this is that it doesn't tell you where in the read stream bytes were dropped, just that they were dropped. I think to get something like "X characters were dropped here" in the stream would require a big refactor, may not be worth it if this failure mode can be mostly avoided.